### PR TITLE
Add JAX Slab Correction Bindings

### PR DESCRIPTION
**Pull request by atulcthakur**
# ALCHEMI Toolkit-Ops Pull Request

## Description

Adds explicit-output JAX support for 2D electrostatic slab corrections, matching the Torch Ewald/PME slab semantics. JAX autograd is not enabled to maintain consistency with Ewald/PME bindings. 

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Performance improvement
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] CI/CD or infrastructure change

## Changes Made

- Added JAX `compute_slab_correction(...)` with explicit energies, forces, charge gradients, and virials.
- Extended JAX Ewald and PME wrappers with `pbc` / `slab_correction` support and shared output-composition helpers.
- Added JAX slab tests covering standalone correction, Ewald/PME composition, output ordering, dtype behavior, and `pbc` shape contracts.

## Testing

- [x] Unit tests pass locally (`make pytest`)
- [x] Linting passes (`make lint`)
- [x] New tests added for new functionality meets coverage expectations?

## Checklist

- [x] I have read and understand the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] I have updated the [CHANGELOG.md](../CHANGELOG.md)
- [x] I have performed a self-review of my code
- [x] I have added docstrings to new functions/classes
- [ ] I have updated the documentation (if applicable)

## Additional Notes

> [!TIP]
> This repository uses Greptile, an AI code review service, to help conduct
> pull request reviews. We encourage contributors to read and consider suggestions
> made by Greptile, but note that human maintainers will provide the necessary
> reviews for merging: Greptile's comments are **not** a qualitative judgement
> of your code, nor is it an indication that the PR will be accepted/rejected.
> We encourage the use of emoji reactions to Greptile comments, depending on
> their usefulness and accuracy.

### CODE REVIEW

**Comment by atulcthakur** [comment]
/ok to test https://github.com/NVIDIA/nvalchemi-toolkit-ops/pull/90/commits/69e23a207c0fa0f1138f3781e533c616bf5b9684

**Comment by atulcthakur** [comment]
/ok to test https://github.com/NVIDIA/nvalchemi-toolkit-ops/pull/90/commits/d00a6d22f8b76d49c4d492f29fae2be0909f5e58

File: nvalchemiops/jax/interactions/electrostatics/_utils.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+from warp.jax_experimental import jax_kernel
+
+
+class ElectrostaticOutputs(NamedTuple):
+    """Named electrostatic outputs used for internal composition."""
+
+    energies: jax.Array
+    forces: jax.Array | None = None
+    charge_grads: jax.Array | None = None
+    virial: jax.Array | None = None
```
Comment (dallasfoster) [inline nvalchemiops/jax/interactions/electrostatics/_utils.py]: I'm just not sure I'm particularly fond of this device - it doesn't really match any of the other user facing code.

File: nvalchemiops/jax/interactions/electrostatics/slab.py [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+    positions_cast = positions.astype(dtype)
+    charges_cast = charges.astype(dtype)
+    cell_cast, num_systems = _prepare_cell(cell.astype(dtype))
+    pbc_cast = _prepare_pbc_for_slab(pbc, num_systems)
+    num_atoms = positions_cast.shape[0]
+
+    if batch_idx is None:
+        batch_idx_i32 = jnp.zeros(num_atoms, dtype=jnp.int32)
+    else:
+        batch_idx_i32 = batch_idx.astype(jnp.int32)
```
Comment (dallasfoster) [inline nvalchemiops/jax/interactions/electrostatics/slab.py:158]: So the consequence here is that if batch_idx is None then we automatically allocate an array and this method is not graphable - is that right?

Comment (dallasfoster) [reply nvalchemiops/jax/interactions/electrostatics/slab.py:158]: I see other allocations below, maybe I don't understand how jax traces. @zubatyuk is there any concern here about performance?

Comment (zubatyuk) [reply nvalchemiops/jax/interactions/electrostatics/slab.py:158]: Not really. But we usually give an option to supply buffer and avoid allocations.

File: benchmarks/interactions/electrostatics/benchmark_electrostatics.py [hunk 3]
```diff
@@ -43,6 +43,8 @@
     python benchmark_electrostatics.py --config benchmark_config.yaml --backend torchpme --method ewald
     python benchmark_electrostatics.py --config benchmark_config.yaml --backend torch --method ewald_slab
     python benchmark_electrostatics.py --config benchmark_config.yaml --backend torch --method pme_slab
+    python benchmark_electrostatics.py --config benchmark_config.yaml --backend jax --method ewald_slab
+    python benchmark_electrostatics.py --config benchmark_config.yaml --backend jax --method pme_slab
```
Comment (zubatyuk) [inline benchmarks/interactions/electrostatics/benchmark_electrostatics.py:47]: `--method all --backend both` will not benchmark the new JAX slab paths

File: benchmarks/interactions/electrostatics/benchmark_electrostatics.py [hunk 4]
```diff
@@ -637,6 +639,41 @@ def prepare_batch_system(
...
 
 
+def prepare_jax_ewald_pme_system(np_data: dict, dtype_str: str) -> dict:
+    """Prepare a JAX system dictionary for Ewald and PME benchmarks."""
+    backend_data = convert_to_backend(np_data, "jax", dtype_str=dtype_str)
+    params_data = compute_electrostatics_params(backend_data, "jax")
+    nl_matrix, nl_num_neighbors, nl_matrix_shifts = compute_neighbor_list(
+        backend_data, "jax", params_data["cutoff"]
+    )
+    pbc_slab = backend_data["pbc"].at[..., 2].set(False)
```
Comment (zubatyuk) [inline benchmarks/interactions/electrostatics/benchmark_electrostatics.py]: `"pbc"` changes after NL is computed. Do we have inconsistent NL?

Comment (zubatyuk) [inline benchmarks/interactions/electrostatics/benchmark_electrostatics.py]: Should we build a slab system, instead of crystal?

Comment (atulcthakur) [reply benchmarks/interactions/electrostatics/benchmark_electrostatics.py]: Great catch

**Comment by atulcthakur** [comment]
/ok to test https://github.com/NVIDIA/nvalchemi-toolkit-ops/pull/90/commits/b44ca501bfa3d74b782d67e9ee002ee300051c9e

**Comment by atulcthakur** [comment]
/ok to test https://github.com/NVIDIA/nvalchemi-toolkit-ops/pull/90/commits/739ae1c6186a1551bee1fd97d0dccf99cb82aaac

File: benchmarks/interactions/electrostatics/benchmark_electrostatics.py (hunk 3 above)
Comment (atulcthakur) [reply benchmarks/interactions/electrostatics/benchmark_electrostatics.py:47]: This is intentional and consistent with other JAX Ewald/PME paths

File: benchmarks/interactions/electrostatics/benchmark_electrostatics.py (hunk 4 above)
Comment (atulcthakur) [reply benchmarks/interactions/electrostatics/benchmark_electrostatics.py]: Now builds a slab system

**Review by dallasfoster — APPROVED** [review]

**Comment by dallasfoster** [comment]
/ok to test [13d1d4a](https://github.com/NVIDIA/nvalchemi-toolkit-ops/pull/90/commits/13d1d4a6e341b7a5163ad8a728250bc5f2ebdf03)
